source/backend/dal/Repositories/PropertyActivityRepository.cs

```diff
@@ -142,6 +160,29 @@ public bool TryDelete(long activityId)
             return deletedSuccessfully;
         }
 
+        /// <summary>
+        /// Delete an activity, and all property-activity relationships.
+        /// </summary>
+        /// <param name="activityId"></param>
+        /// <returns>Boolean of deletion sucess.</returns>
+        public bool TryDeleteFile(long activityId, long managementFileId)
+        {
+            // There may be zero to many of these relationships.
+            Context.PimsPropPropActivities.ForEach(pp =>
+            {
+                Context.PimsPropPropActivities.Remove(pp);
+            });
```

Comment:
hmm - this sounds dangerous. If I'm reading this right, it would delete **ALL** property-activity associations (not just the ones between the desired activity and its properties

Maybe something along the lines of:
```suggestion
            Context.PimsPropPropActivities.Where(x => x.PimsPropertyActivityId == activityId).ForEach(pp =>
            {
                Context.PimsPropPropActivities.Remove(pp);
            });
```